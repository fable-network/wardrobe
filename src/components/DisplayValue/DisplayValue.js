import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ellipsis = `
  overflow: hidden;
  white-space: nowrap;  
  text-overflow: ellipsis;
`;

const Label = styled.div`
  max-width: 100%;
  ${ellipsis};
  font-weight: ${p => p.theme.fontWeightBold};
`;

const Content = styled.div`
  font-size: ${p => p.theme.fontSizeLarge};
  line-height: ${p => p.theme.lineHeightLarge};
  color: ${p => p.theme.grey02};
`;

const Wrapper = styled.div`
  display: flex;
  flex-flow: column nowrap;
  
  font-family: inherit;
  font-weight: ${p => p.theme.fontWeightNormal};
  font-size: ${p => p.theme.fontSizeSmall};
  line-height: ${p => p.theme.lineHeightSmall};
`;

const DisplayValue = ({ label, children }) => (
  <Wrapper>
    <Label>{label}</Label>
    <Content>{children}</Content>
  </Wrapper>
);

DisplayValue.propTypes = {
  label: PropTypes.string.isRequired,
  children: PropTypes.node,
};

export default DisplayValue;
