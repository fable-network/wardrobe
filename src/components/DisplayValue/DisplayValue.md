
DisplayValue component to render a given value with a label on top:

```jsx
<DisplayValue label="Total WSP">
EUR <span style={{textDecoration:'line-through'}}>15,982.09</span> <span style={{color:'red'}}>12,345.67</span>
</DisplayValue> 
```

With not too much content:

```jsx
<DisplayValue label="Pieces">0</DisplayValue>
```
