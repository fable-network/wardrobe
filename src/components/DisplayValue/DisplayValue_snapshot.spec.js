/* eslint-disable */
// This file was automatically generated
import React from 'react';
import { shallow } from 'enzyme';
import DisplayValue from './DisplayValue';

describe('DisplayValue Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = shallow(
        <DisplayValue label="Total WSP">
        EUR <span style={{textDecoration:'line-through'}}>15,982.09</span> <span style={{color:'red'}}>12,345.67</span>
        </DisplayValue>
      );
      expect(wrapper1).toMatchSnapshot();

      const wrapper2 = shallow(
        <DisplayValue label="Pieces">0</DisplayValue>
      );
      expect(wrapper2).toMatchSnapshot();

    });
  });
});
