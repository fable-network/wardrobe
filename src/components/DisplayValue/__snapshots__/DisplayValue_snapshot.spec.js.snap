// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayValue Component snapshots should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DisplayValue
    label="Total WSP"
  >
    EUR 
    <span
      style={
        Object {
          "textDecoration": "line-through",
        }
      }
    >
      15,982.09
    </span>
     
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      12,345.67
    </span>
  </DisplayValue>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <DisplayValue__Label>
          Total WSP
        </DisplayValue__Label>,
        <DisplayValue__Content>
          EUR 
          <span
            style={
              Object {
                "textDecoration": "line-through",
              }
            }
          >
            15,982.09
          </span>
           
          <span
            style={
              Object {
                "color": "red",
              }
            }
          >
            12,345.67
          </span>
        </DisplayValue__Content>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Total WSP",
        },
        "ref": null,
        "rendered": "Total WSP",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "EUR ",
            <span
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              15,982.09
            </span>,
            " ",
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
              12,345.67
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "EUR ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "15,982.09",
              "style": Object {
                "textDecoration": "line-through",
              },
            },
            "ref": null,
            "rendered": "15,982.09",
            "type": "span",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "12,345.67",
              "style": Object {
                "color": "red",
              },
            },
            "ref": null,
            "rendered": "12,345.67",
            "type": "span",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <DisplayValue__Label>
            Total WSP
          </DisplayValue__Label>,
          <DisplayValue__Content>
            EUR 
            <span
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              15,982.09
            </span>
             
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
              12,345.67
            </span>
          </DisplayValue__Content>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Total WSP",
          },
          "ref": null,
          "rendered": "Total WSP",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "EUR ",
              <span
                style={
                  Object {
                    "textDecoration": "line-through",
                  }
                }
              >
                15,982.09
              </span>,
              " ",
              <span
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                12,345.67
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "EUR ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "15,982.09",
                "style": Object {
                  "textDecoration": "line-through",
                },
              },
              "ref": null,
              "rendered": "15,982.09",
              "type": "span",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "12,345.67",
                "style": Object {
                  "color": "red",
                },
              },
              "ref": null,
              "rendered": "12,345.67",
              "type": "span",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`DisplayValue Component snapshots should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DisplayValue
    label="Pieces"
  >
    0
  </DisplayValue>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <DisplayValue__Label>
          Pieces
        </DisplayValue__Label>,
        <DisplayValue__Content>
          0
        </DisplayValue__Content>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Pieces",
        },
        "ref": null,
        "rendered": "Pieces",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "0",
        },
        "ref": null,
        "rendered": "0",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <DisplayValue__Label>
            Pieces
          </DisplayValue__Label>,
          <DisplayValue__Content>
            0
          </DisplayValue__Content>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Pieces",
          },
          "ref": null,
          "rendered": "Pieces",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "0",
          },
          "ref": null,
          "rendered": "0",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
